app/helpers/user_controller_helper.rb

```diff
@@ -6,4 +6,16 @@ def render_collaborators_for(user)
   def render_demo_logout_button_for(user)
     render "users/demo_logout_button" if user.is_demo_user?
   end
+
+  def resource_name
+    :user
+  end
+
+  def resource
+    @user
+  end
+
+  def devise_mapping
+    @devise_mapping ||= Devise.mappings[:user]
+  end
```

Comment:
Is the `resource_name` method used anywhere? Also the `resource` method, I don't see those being used. Is this devise magic? Must be devise magic. No other reason to cache a `@devise_mapping` instance variable.